### PR TITLE
fix: Improvements to Organization model setup

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Found some small issues whilst looking at Organization levels

## Changes

* Changes the default decision to be done at runtime rather than at the DB level
* Tightens up some params of the API

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Added tests

### CODE REVIEW

File: posthog/migrations/0272_alter_organization_plugins_access_level.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('posthog', '0271_delete_promptsequencestate'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='organization',
+            name='plugins_access_level',
+            field=models.PositiveSmallIntegerField(choices=[(0, 'none'), (3, 'config'), (6, 'install'), (9, 'root')], default=3),
```
Comment (benjackwhite) [inline posthog/migrations/0272_alter_organization_plugins_access_level.py:16]: This should be a safe change as this was anyway the default for cloud and for self-hosted this is only affects new installations

File: posthog/migrations/0272_alter_organization_plugins_access_level.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0271_delete_promptsequencestate"),
+    ]
+
+    operations = [
+        migrations.AlterField(
```
Comment (pauldambra) [inline posthog/migrations/0272_alter_organization_plugins_access_level.py:13]: would be interesting to look at the table DDL... My guess is this is a no-op for the DB and only helps Django. Nice to add that as a comment to migrations so we know whether to care about migration performance

**Review by pauldambra — APPROVED** [review]
Not run it but looks sensible to me!

File: posthog/migrations/0272_alter_organization_plugins_access_level.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/migrations/0272_alter_organization_plugins_access_level.py:13]: I think you're right. It isn't actually modifying anything (my local DB has no default)

**Review by mariusandra — APPROVED** [review]
Looks sensible

**Review by Twixes — APPROVED** [review]

**Pull request merged**